### ISSUE TITLE
Remove arbitrary character limit in posts & alt text or at least make these configurable settings

### ISSUE THREAD

**Issue by SleepyCatten** [mastodon/mastodon#24147]
### Pitch

Mastodon instances currently have a post character limit of 500 characters, which seems to only be something that servers can work around by using Mastodon forks, like glitch-soc, or different Fediverse software like Misskey or Calckey.

Additionally, this character limit includes the text in a content warning, custom emojis and hashtags.

For alt text, the limit is a bit higher at 1500 characters. However, as can be seen from existing separate requests, this limit is not enough for full text descriptions of many images. It is only 50% higher than the very restrictive alt text limit on Twitter and seems to be rather arbitrary, rather than something required.

### Motivation

Right now, it is very difficult for users on Mastodon instances to make detailed posts. These can be manually done by making threaded posts, but there is no native functionality to make threads, so each post must be done in reply to the one above, one at a time. In addition to making it more difficult for posters to convey their message, it also clogs up the timelines of their followers.

Removing the character limit, but perhaps reducing the immediate preview of posts to 500, which users can then expand to view the entire post, would be a welcome change to Mastodon users who want to write longer posts without constantly trying to work around the restrictive character limit.

Some might argue that it would reduce the quality of posts or that the restrictive character limit is there to encourage ingenuity. However, this argument can be shown to be demonstratively incorrect by:

1. Reading posts on Twitter with an even shorter character post limit. (There's little ingenuity there.)
2. Seeing how many threaded posts exist on Mastodon posts to have to work around the limit.
3. Seeing the quality of longer posts by users of Mastodon forks or other Fediverse software.

It will:

1. Improve the wellbeing of users who struggle to convey thoughts concisely.
2. Help with accessibility, as it will make it both easier to post & to read (or have read with a screen reader) long threaded posts, plus describe images with text in full.
3. Help set Mastodon further apart from Twitter.
4. Allow for more descriptive custom emojis, which will help people with screen readers.

I'm sure that there are other benefits too, but these are the ones that sprung to mind.

**Comment by cloo** [comment]
Doubling down. This is not only about "expressing thoughts concisely". Stories are things to share, why make them difficult in mastodon? Also, threads are a terrible user experience, especially when they show up scattered or even reversed. 
It makes no sense to make this tribute to sms heritage of ex twitter.

**Comment by expertmanofficial** [comment]
Agreed. Subcribing to that issue. I'm tired of finding instances that would allow me to have higher character count.

**Comment by Cassolotl** [comment]
See also *Let the admins easily setup the character limits #12265* (October 2019, already 4 years old! Ouch.)

In 2018 Gargron said that "character limit is not planned to be configurable". https://github.com/mastodon/mastodon/issues/5693#issuecomment-389934011

**Comment by SleepyCatten** [comment]
> See also _Let the admins easily setup the character limits #12265_ (October 2019, already 4 years old! Ouch.)
> 
> In 2018 Gargron said that "character limit is not planned to be configurable". [#5693 (comment)](https://github.com/mastodon/mastodon/issues/5693#issuecomment-389934011)

Hopefully they'll change their mind. 2918 and 2019 were both a lifetime ago in software years. I don't imagine the Mastodon devs ever expected Twitter to implode and Mastodon to increase so much in popularity. Hopefully they'll be more receptive now.

**Comment by vmstan** [comment]
Closed as duplicate of #12265

**Comment by SleepyCatten** [comment]
> Closed as duplicate of #12265

This is the WHOLE point of raising issues and suggestions: to ADD them to planning.

It's not a duplicate either.

**Comment by rkingett** [comment]
Agreed! It's becoming tedious to help people find instances with a higher character limit so they don't have to worry about providing accessibility. The alt text character limit should be removed!
